--- CMakeLists.txt
+++ CMakeLists.txt
@@ -418,6 +418,10 @@
     set(boost_components "")
     set(boost_components ${boost_components} program_options)
     find_package( Boost 1.82 COMPONENTS ${boost_components})
+    if (WIN32 AND NOT STATICCOMPILE)
+        add_definitions(-DBOOST_ALL_NO_LIB)
+        add_definitions(-DBOOST_ALL_DYN_LINK)
+    endif()
 endif()

 option(STATS "Don't use statistics at all" OFF)
